tooling/nargo/src/artifacts/debug_vars.rs

```diff
@@ -0,0 +1,132 @@
+            .unwrap_or_else(|| panic!("value unavailable for var_id {var_id}"));
+        let cursor_type_id = self
+            .id_to_type
+            .get(&var_id)
+            .unwrap_or_else(|| panic!("type id unavailable for var_id {var_id}"));
+        let mut cursor_type = self
+            .types
+            .get(cursor_type_id)
+            .unwrap_or_else(|| panic!("type unavailable for type id {cursor_type_id}"));
+        for index in indexes.iter() {
+            (cursor, cursor_type) = match (cursor, cursor_type) {
+                (PrintableValue::Vec(array), PrintableType::Array { length, typ }) => {
+                    if let Some(len) = length {
+                        if *index as u64 >= *len {
+                            panic!("unexpected field index past array length")
+                        }
+                        if *len != array.len() as u64 {
+                            panic!("type/array length mismatch")
+                        }
+                    }
+                    (array.get_mut(*index as usize).unwrap(), &*Box::leak(typ.clone()))
+                }
+                (
+                    PrintableValue::Struct(field_map),
+                    PrintableType::Struct { name: _name, fields },
+                ) => {
+                    if *index as usize >= fields.len() {
+                        panic!("unexpected field index past struct field length")
+                    }
+                    let (key, typ) = fields.get(*index as usize).unwrap();
+                    (field_map.get_mut(key).unwrap(), typ)
+                }
+                (PrintableValue::Vec(array), PrintableType::Tuple { types }) => {
+                    if *index >= types.len() as u32 {
+                        panic!(
+                            "unexpected field index ({index}) past tuple length ({})",
+                            types.len()
+                        );
+                    }
+                    if types.len() != array.len() {
+                        panic!("type/array length mismatch")
+                    }
+                    let typ = types.get(*index as usize).unwrap();
+                    (array.get_mut(*index as usize).unwrap(), typ)
+                }
+                _ => {
+                    panic!("unexpected assign field of {cursor_type:?} type");
+                }
+            };
+        }
+        *cursor = decode_value(&mut values.iter().map(|v| v.to_field()), cursor_type);
+        self.active.insert(var_id);
+    }
+
+    pub fn assign_deref(&mut self, _var_id: u32, _values: &[Value]) {
+        // TODO
+        unimplemented![]
+    }
+
+    pub fn get(&mut self, var_id: u32) -> Option<&PrintableValue> {
+        self.id_to_value.get(&var_id)
+    }
+
+    pub fn get_type(&self, var_id: u32) -> Option<&PrintableType> {
+        self.id_to_type.get(&var_id).and_then(|type_id| self.types.get(type_id))
+    }
+
+    pub fn drop(&mut self, var_id: u32) {
```

Comment:
Can you rename this to `drop_var` or `remove_var`? When I read `drop` it sounds like we are implementing `Drop` and dropping the entire struct.